core/unit_test/TestViewLayoutStrideAssignment.hpp

```diff
@@ -768,6 +933,161 @@ TEST(TEST_CATEGORY_DEATH, view_layoutstride_right_to_layoutleft_assignment) {
   }
 }
 
+TEST(TEST_CATEGORY, view_layoutstride_left_to_layoutright_assignable) {
+  typedef TEST_EXECSPACE exec_space;
+
+  auto t = time(nullptr);
+  srand(t);  // Use current time as seed for random generator
+  printf("view_layoutstride_left_to_layoutright_assignment: srand(%lu)\n",
+         size_t(t));
+
+  {  // Assignment of rank-1 LayoutRight = LayoutStride (LayoutLeft compatible)
+    int ndims   = 1;
+    int dims[]  = {10};
+    int order[] = {0};
+    Kokkos::LayoutStride layout =
+        Kokkos::LayoutStride::order_dimensions(ndims, order, dims);
+    Kokkos::View<double*, Kokkos::LayoutStride, exec_space> src("LayoutStride",
+                                                                layout);
+
+    Kokkos::View<double*, Kokkos::LayoutStride, exec_space>::HostMirror h_src =
+        Kokkos::create_mirror_view(src);
+
+    for (size_t i = 0; i < src.span(); i++)
+      h_src.data()[i] = (double)rand() / RAND_MAX * (100);
+
+    Kokkos::deep_copy(src, h_src);
+
+    Kokkos::View<double*, Kokkos::LayoutRight, exec_space> dst;
+
+    if (dst.is_assignable(src)) {
+      dst = src;
+
+      Kokkos::View<double*, Kokkos::LayoutRight, exec_space>::HostMirror h_dst =
+          Kokkos::create_mirror_view(dst);
+
+      Kokkos::deep_copy(h_dst, dst);
+
+      bool test = true;
+      for (size_t i = 0; i < src.span(); i++) {
+        if (h_src.data()[i] != h_dst.data()[i]) {
+          test = false;
+          break;
+        }
+      }
+      ASSERT_EQ(dst.span(), src.span());
+      ASSERT_EQ(test, true);
+    } else {
+      ASSERT_TRUE(false);
+    }
+  }
+  {  // Assignment of rank-2 LayoutRight = LayoutStride (LayoutLeft compatible)
+    int ndims   = 2;
+    int dims[]  = {10, 9};
+    int order[] = {0, 1};
+    Kokkos::LayoutStride layout =
+        Kokkos::LayoutStride::order_dimensions(ndims, order, dims);
+    Kokkos::View<double**, Kokkos::LayoutStride, exec_space> src("LayoutStride",
+                                                                 layout);
+
+    Kokkos::View<double**, Kokkos::LayoutRight, exec_space> dst;
+
+    if (dst.is_assignable(src)) {
+      ASSERT_TRUE(false);
+    }
```

Comment:
```suggestion
    ASSERT_FALSE(dst.is_assignable(src))
```

(And all of the other places below)